src/scipp/core/data_group.py

```diff
@@ -0,0 +1,192 @@
+            for name, item in items.items():
+                self[name] = item
+
+    def __len__(self) -> int:
+        return len(self._items)
+
+    def __iter__(self):
+        yield from self._items
+
+    def __getitem__(self, name):
+        if isinstance(name, str):
+            return self._items[name]
+        if name == ():
+            return DataGroup({key: var[()] for key, var in self.items()})
+        dim, index = name
+        if isinstance(index, int) and self.sizes[dim] is None:
+            raise ValueError(
+                f"Positional indexing dim {dim} not possible as the length is not "
+                "unique.")
+        return DataGroup({
+            key: var[dim, index] if dim in _item_dims(var) else var
+            for key, var in self.items()
+        })
+
+    def __setitem__(self, name, value):
+        self._items[name] = value
+
+    def __delitem__(self, name: str):
+        del self._items[name]
+
+    @property
+    def dims(self):
+        dims = ()
+        for var in self.values():
+            # Preserve insertion order
+            for dim in _item_dims(var):
+                if dim not in dims:
+                    dims = dims + (dim, )
+        return dims
+
+    @property
+    def ndim(self):
+        return len(self.dims)
+
+    @property
+    def shape(self):
+
+        def dim_size(dim):
+            sizes = []
+            for var in self.values():
+                if dim in _item_dims(var):
+                    sizes.append(var.sizes[dim])
+            sizes = set(sizes)
+            if len(sizes) == 1:
+                return next(iter(sizes))
+            return None
+
+        return tuple(dim_size(dim) for dim in self.dims)
+
+    @property
+    def sizes(self):
+        return dict(zip(self.dims, self.shape))
+
+    def _make_html(self):
+        out = ''
+        for name, item in self.items():
+            if isinstance(item, DataGroup):
+                html = item._make_html()
+            elif isinstance(item, (Variable, DataArray, Dataset)):
+                html = ''
+            else:
+                html = str(item)
+            out += f"<details style=\"padding-left:2em\"><summary>"\
+                   f"{name}: {_summarize(item)}</summary>{html}</details>"
+        return out
+
+    def _repr_html_(self):
+        out = ''
+        out += f"<details open=\"open\"><summary>DataGroup"\
+               f"({len(self)})</summary>"
+        out += self._make_html()
+        out += "</details>"
+        return out
+
+    def __repr__(self):
+        r = 'DataGroup(\n'
+        for name, var in self.items():
+            r += f'    {name}: {_summarize(var)}\n'
+        r += ')'
+        return r
+
+    def __eq__(self, other):
+        return _data_group_binary(operator.eq, self, other)
+
+    def __add__(self, other):
+        return _data_group_binary(operator.add, self, other)
+
+    def __mul__(self, other):
+        return _data_group_binary(operator.mul, self, other)
```

Comment:
Why only those two operators? Could at least have minus and divide?

Reply:
Because it was not in the prototype. Will add more things later. See the PR description.